plugins/out_kinesis_firehose/firehose_api.c

```diff
@@ -216,8 +217,8 @@ static int process_event(struct flb_firehose *ctx, struct flush *buf,
                          ctx->delivery_stream);
             return 2;
         }
-        /* guess space needed to write time_key */
-        len = 6 + strlen(ctx->time_key) + 6 * strlen(ctx->time_key_format);
+        /* format time output and get the length */
+        len = create_time_format_string(&out_buff, ctx->time_key_format, tms);
```

Comment:
What if this function returns a negative (-1) for error. Don't you need code to handle this?